target_s3/formats/format_base.py

```diff
@@ -58,6 +58,7 @@ def __init__(self, config: dict, context: dict, extension: str) -> None:
             self.session = Session(
                 aws_access_key_id=aws_config.get("aws_access_key_id", None),
                 aws_secret_access_key=aws_config.get("aws_secret_access_key", None),
+                aws_session_token=aws_config.get("aws_session_token", None),
```

Comment:
@Vzzarr -- I'm not super familiar, this `aws_session_token` is something we manually pass in? The way this is configured now, I think we also need to add a property input here: https://github.com/crowemi/target-s3/blob/main/target_s3/target.py#L65-L102

And, update the example config: https://github.com/crowemi/target-s3/blob/main/sample-config.json#L12-L19

Reply:
Quoting the [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html):

> **aws_session_token (string)** – The session token to use when creating the client. Same semantics as aws_access_key_id above.

So this is optional but some AWS Configurations require this session token to allow connection to the AWS Account, in these cases this would be really helpful.
I'm using my forked repo and this is working for me.

Also added the required changes.

One more thing I wanted to raise, I configured this target via Meltano and I had to create the following:
```
    config:
      prefix: $S3_KEY
      format:
        format_type: json
      cloud_provider:
        cloud_provider_type: aws
        aws:
          aws_access_key_id: $AWS_ACCESS_KEY_ID
          aws_secret_access_key: $AWS_SECRET_ACCESS_KEY
          aws_region: $AWS_DEFAULT_REGION
          aws_session_token: $AWS_SESSION_TOKEN
          aws_bucket: $S3_BUCKET
```

But using the `meltano config target-s3 set aws_access_key $AWS_ACCESS_KEY_ID` as per https://hub.meltano.com/loaders/target-s3/ it didn't add params like `cloud_provider`, `cloud_provider_type` or `aws`. 
I had to manually add them after many tests and reading through the code base.
But maybe this can be addressed in a different PR, just wanted to raise my experience to your attention.